XamarinCommunityToolkit/Effects/SafeAreaEffectRouter.ios.cs

```diff
@@ -24,24 +24,27 @@ protected override void OnAttached()
 
 			var insets = UIApplication.SharedApplication.Windows[0].SafeAreaInsets;
 
-			initialMargin = element.Margin;
-
 			if (insets.Top <= 0)
 				return;
 
+			initialMargin = element.Margin;
+
 			var safeArea = SafeAreaEffect.GetSafeArea(element);
 
 			element.Margin = new Thickness(
-				initialMargin.Left + (safeArea.Left ? insets.Left : 0),
-				initialMargin.Top + (safeArea.Top ? insets.Top : 0),
-				initialMargin.Right + (safeArea.Right ? insets.Right : 0),
-				initialMargin.Bottom + (safeArea.Bottom ? insets.Bottom : 0));
+				element.Margin.Left + (safeArea.Left ? insets.Left : 0),
+				element.Margin.Top + (safeArea.Top ? insets.Top : 0),
+				element.Margin.Right + (safeArea.Right ? insets.Right : 0),
+				element.Margin.Bottom + (safeArea.Bottom ? insets.Bottom : 0));
 		}
 
 		protected override void OnDetached()
 		{
-			if (Element is Layout element)
```

Comment:
```csharp
if (isApplied)
    element.Margin = initialMargin;

isApplied = false;
```

Reply:
Why? Nullable value types exist exactly for avoiding that extra state flag.

Reply:
Well, we can leave nullable, but I don't think we should check "Element is Layout"
Just leave "if initialMargin.HasValue"